src/test/scala/io/iohk/ethereum/ledger/LedgerTestSetup.scala

```diff
@@ -400,16 +400,14 @@ trait MockBlockchain extends MockFactory { self: TestSetupWithVmAndValidators =>
       .once()
   }
 
-  def setHeaderByHash(hash: ByteString, header: Option[BlockHeader]): CallHandler1[ByteString, Option[BlockHeader]] =
-    (blockchain.getBlockHeaderByHash _).expects(hash).returning(header)
+  def setHeaderInChain(hash: ByteString, result: Boolean = true): CallHandler1[ByteString, Boolean] =
```

Comment:
looking at `setGenesisHeader` why is `setHeaderInChain` returning `CallHandler1[ByteString, Boolean]` and the former is returning `Unit`?

Reply:
setHeaderInChain expects 1 parameter while setGenesisheader has no parameters